The job **`x86_64-gnu-llvm-19`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14934613321/job/41959002809) [(plain)](https://github.com/rust-lang/rust/commit/705eda2267e55bde4fe96fe61a03b62f6c9aa1f0/checks/41959002809/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#18 exporting to docker image format
#18 sending tarball 20.4s done
#18 DONE 27.6s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-19]
[CI_JOB_NAME=x86_64-gnu-llvm-19]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Listening on address 127.0.0.1:4226
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-19', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'build.print-step-timings', '--enable-verbose-tests', '--set', 'build.metrics', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--set', 'gcc.download-ci-gcc=true', '--enable-new-symbol-mangling']
configure: build.build          := x86_64-unknown-linux-gnu
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-19/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
failures:

---- [assembly] tests/assembly/asm/aarch64-types.rs#arm64ec stdout ----

error in revision `arm64ec`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-19/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/asm/aarch64-types.arm64ec/aarch64-types.s" "/checkout/tests/assembly/asm/aarch64-types.rs" "--check-prefix=CHECK" "--check-prefix" "arm64ec" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/assembly/asm/aarch64-types.rs:89:11: error: CHECK: expected string not found in input
// CHECK: str {{.*}}x30
          ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/asm/aarch64-types.arm64ec/aarch64-types.s:63:21: note: scanning from here
 .def "#issue_75761";
                    ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/asm/aarch64-types.arm64ec/aarch64-types.s:74:2: note: possible intended match here
 stp x29, x30, [sp, #-16]!
 ^

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/asm/aarch64-types.arm64ec/aarch64-types.s
Check file: /checkout/tests/assembly/asm/aarch64-types.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
            1:  .text 
            2:  .def "@feat.00"; 
            3:  .scl 3; 
            4:  .type 0; 
            5:  .endef 
            6:  .globl "@feat.00" 
            7: .set "@feat.00", 0 
            8:  .file "aarch64_types.4463f54976dfa885-cgu.0" 
            9:  .def "#sym_fn"; 
           10:  .scl 2; 
           11:  .type 32; 
           12:  .endef 
           13:  .section .text,"xr",discard,"#sym_fn" 
           14:  .globl "#sym_fn" 
           15:  .p2align 2 
           16: "#sym_fn": 
           17:  .weak_anti_dep sym_fn 
           18: .set sym_fn, "#sym_fn" 
           19: .seh_proc "#sym_fn" 
           20:  stp x29, x30, [sp, #-16]! 
           21:  .seh_save_fplr_x 16 
           22:  mov x29, sp 
           23:  .seh_set_fp 
           24:  .seh_endprologue 
           25:  //APP 
           26:  bl extern_func 
           27:  //NO_APP 
           28:  .seh_startepilogue 
           29:  ldp x29, x30, [sp], #16 
           30:  .seh_save_fplr_x 16 
           31:  .seh_endepilogue 
           32:  ret 
           33:  .seh_endfunclet 
           34:  .seh_endproc 
           35:  
           36:  .def "#sym_static"; 
           37:  .scl 2; 
           38:  .type 32; 
           39:  .endef 
           40:  .section .text,"xr",discard,"#sym_static" 
           41:  .globl "#sym_static" 
           42:  .p2align 2 
           43: "#sym_static": 
           44:  .weak_anti_dep sym_static 
           45: .set sym_static, "#sym_static" 
           46: .seh_proc "#sym_static" 
           47:  stp x29, x30, [sp, #-16]! 
           48:  .seh_save_fplr_x 16 
           49:  mov x29, sp 
           50:  .seh_set_fp 
           51:  .seh_endprologue 
           52:  //APP 
           53:  adr x0, extern_static 
           54:  //NO_APP 
           55:  .seh_startepilogue 
           56:  ldp x29, x30, [sp], #16 
           57:  .seh_save_fplr_x 16 
           58:  .seh_endepilogue 
           59:  ret 
           60:  .seh_endfunclet 
           61:  .seh_endproc 
           62:  
           63:  .def "#issue_75761"; 
check:89'0                         X~ error: no match found
           64:  .scl 2; 
check:89'0     ~~~~~~~~~
           65:  .type 32; 
check:89'0     ~~~~~~~~~~~
           66:  .endef 
check:89'0     ~~~~~~~~
           67:  .section .text,"xr",discard,"#issue_75761" 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           68:  .globl "#issue_75761" 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~~~
           69:  .p2align 2 
check:89'0     ~~~~~~~~~~~~
           70: "#issue_75761": 
check:89'0     ~~~~~~~~~~~~~~~~
           71:  .weak_anti_dep issue_75761 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           72: .set issue_75761, "#issue_75761" 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           73: .seh_proc "#issue_75761" 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
           74:  stp x29, x30, [sp, #-16]! 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:89'1      ?                          possible intended match
           75:  .seh_save_fplr_x 16 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~
           76:  mov x29, sp 
check:89'0     ~~~~~~~~~~~~~
           77:  .seh_set_fp 
check:89'0     ~~~~~~~~~~~~~
           78:  .seh_endprologue 
check:89'0     ~~~~~~~~~~~~~~~~~~
           79:  //APP 
check:89'0     ~~~~~~~
           80:  //NO_APP 
check:89'0     ~~~~~~~~~~
           81:  .seh_startepilogue 
check:89'0     ~~~~~~~~~~~~~~~~~~~~
           82:  ldp x29, x30, [sp], #16 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
           83:  .seh_save_fplr_x 16 
check:89'0     ~~~~~~~~~~~~~~~~~~~~~
           84:  .seh_endepilogue 
check:89'0     ~~~~~~~~~~~~~~~~~~
           85:  ret 
check:89'0     ~~~~~
           86:  .seh_endfunclet 
check:89'0     ~~~~~~~~~~~~~~~~~
           87:  .seh_endproc 
check:89'0     ~~~~~~~~~~~~~~
           88:  
check:89'0     ~
           89:  .def "#reg_i8"; 
check:89'0     ~~~~~~~~~~~~~~~
           90:  .scl 2; 
           91:  .type 32; 
           92:  .endef 
           93:  .section .text,"xr",discard,"#reg_i8" 
           94:  .globl "#reg_i8" 
           95:  .p2align 2 
           96: "#reg_i8": 
           97:  .weak_anti_dep reg_i8 
           98: .set reg_i8, "#reg_i8" 
           99: .seh_proc "#reg_i8" 
          100:  stp x29, x30, [sp, #-16]! 
          101:  .seh_save_fplr_x 16 
          102:  mov x29, sp 
          103:  .seh_set_fp 
          104:  .seh_endprologue 
          105:  ldrb w8, [x0] 
          106:  //APP 
          107:  mov x9, x8 
          108:  //NO_APP 
          109:  strb w9, [x1] 
          110:  .seh_startepilogue 
          111:  ldp x29, x30, [sp], #16 
          112:  .seh_save_fplr_x 16 
          113:  .seh_endepilogue 
          114:  ret 
          115:  .seh_endfunclet 
          116:  .seh_endproc 
          117:  
          118:  .def "#reg_i16"; 
          119:  .scl 2; 
          120:  .type 32; 
          121:  .endef 
          122:  .section .text,"xr",discard,"#reg_i16" 
          123:  .globl "#reg_i16" 
          124:  .p2align 2 
          125: "#reg_i16": 
          126:  .weak_anti_dep reg_i16 
          127: .set reg_i16, "#reg_i16" 
          128: .seh_proc "#reg_i16" 
          129:  stp x29, x30, [sp, #-16]! 
          130:  .seh_save_fplr_x 16 
          131:  mov x29, sp 
          132:  .seh_set_fp 
          133:  .seh_endprologue 
          134:  ldrh w8, [x0] 
          135:  //APP 
          136:  mov x9, x8 
          137:  //NO_APP 
          138:  strh w9, [x1] 
          139:  .seh_startepilogue 
          140:  ldp x29, x30, [sp], #16 
          141:  .seh_save_fplr_x 16 
          142:  .seh_endepilogue 
          143:  ret 
          144:  .seh_endfunclet 
          145:  .seh_endproc 
          146:  
          147:  .def "#reg_f16"; 
          148:  .scl 2; 
          149:  .type 32; 
          150:  .endef 
          151:  .section .text,"xr",discard,"#reg_f16" 
          152:  .globl "#reg_f16" 
          153:  .p2align 2 
          154: "#reg_f16": 
          155:  .weak_anti_dep reg_f16 
          156: .set reg_f16, "#reg_f16" 
          157: .seh_proc "#reg_f16" 
          158:  stp x29, x30, [sp, #-16]! 
          159:  .seh_save_fplr_x 16 
          160:  mov x29, sp 
          161:  .seh_set_fp 
          162:  .seh_endprologue 
          163:  ldrh w8, [x0] 
          164:  //APP 
          165:  mov x9, x8 
          166:  //NO_APP 
          167:  strh w9, [x1] 
          168:  .seh_startepilogue 
          169:  ldp x29, x30, [sp], #16 
          170:  .seh_save_fplr_x 16 
          171:  .seh_endepilogue 
          172:  ret 
          173:  .seh_endfunclet 
          174:  .seh_endproc 
            .
            .
            .
>>>>>>
------------------------------------------
```

</details>

--- a/compiler/rustc_target/src/spec/targets/arm64ec_pc_windows_msvc.rs
+++ b/compiler/rustc_target/src/spec/targets/arm64ec_pc_windows_msvc.rs
@@ -1,4 +1,4 @@
-use crate::spec::{LinkerFlavor, Lld, Target, TargetMetadata, add_link_args, base};
+use crate::spec::{FramePointer, LinkerFlavor, Lld, Target, TargetMetadata, add_link_args, base};
 
 pub(crate) fn target() -> Target {
     let mut base = base::windows_msvc::opts();
@@ -9,6 +9,12 @@ pub(crate) fn target() -> Target {
         LinkerFlavor::Msvc(Lld::No),
         &["/machine:arm64ec", "softintrin.lib"],
     );
+
+    // Microsoft recommends enabling frame pointers on Arm64 Windows.
+    // From https://learn.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions?view=msvc-170#integer-registers
+    // "The frame pointer (x29) is required for compatibility with fast stack walking used by ETW
+    // and other services. It must point to the previous {x29, x30} pair on the stack."
+    base.frame_pointer = FramePointer::NonLeaf;
 
     Target {
         llvm_target: "arm64ec-pc-windows-msvc".into(),

--- a/tests/assembly/asm/aarch64-types.rs
+++ b/tests/assembly/asm/aarch64-types.rs
@@ -86,10 +86,12 @@ pub unsafe fn sym_static() {
 
 // Regression test for #75761
 // CHECK-LABEL: {{("#)?}}issue_75761{{"?}}
-// CHECK: str {{.*}}x30
+// aarch64: str {{.*}}x30
+// arm64ec: stp {{.*}}x30
 // CHECK: //APP
 // CHECK: //NO_APP
-// CHECK: ldr {{.*}}x30
+// aarch64: ldr {{.*}}x30
+// arm64ec: ldp {{.*}}x30
 #[no_mangle]
 pub unsafe fn issue_75761() {
     asm!("", out("v0") _, out("x30") _);

